### PR TITLE
Release unshaded version into snowflake repo

### PR DESCRIPTION
**Pull request by sfc-gh-hqin**
In current release of the ingest SDK, we are releasing a shaded jar, which is also called a "fat jar". In the jar file, we includes all the class files from declared dependencies and wiped out the underlying dependency information from the generated pom file. This disabled all the transitive dependency management features provided by maven in order to resolve potential version conflicts. This PR added a script to release unshaded version of the SDK into snowflake maven repo, by declare dependency on the newly added unshaded version, users can use `<exclusion>` tag to exclude problematic transitive dependency.

### CODE REVIEW

File: unshaded_deploy.sh
Comment (sfc-gh-tzhang) [inline unshaded_deploy.sh:3]: This looks a lot similar to deploy.sh, could we consolidate to remove duplicate logic?

File: unshaded_deploy.sh
Comment (sfc-gh-tzhang) [inline unshaded_deploy.sh:65]: I'm curious whether there is a way to test it before merging?

File: unshaded_deploy.sh
Comment (sfc-gh-hqin) [reply unshaded_deploy.sh:3]: Let's maybe keep these two as separated files for now? Reason is:

1. All the other mvn release job run a bash script called `run.sh`, which calls `deploy.sh` under the hood, and `run.sh` does not take any parameter. Modify `deploy.sh` require more changes.
2. Seems no easy way to test the script without deploy to staging repo, so let's better keep the normal release pipeline untouched.

File: unshaded_deploy.sh
Comment (sfc-gh-hqin) [reply unshaded_deploy.sh:65]: I think we can try to release new artifacts into a staging repo in nexus, otherwise not much we can do for testing. @sfc-gh-kdama  can you suggest?

Comment (sfc-gh-japatel) [reply unshaded_deploy.sh:65]: nexus repo sounds like a good idea.But, you can also publish it to local maven repository in your local machine. 
https://www.baeldung.com/maven-local-repository

**Comment by sfc-gh-japatel** [comment]
Can you add a short description of what an unshaded jar is and what are the benefits?

**Review by sfc-gh-tzhang — APPROVED** [review]
LGTM as long as this could be tested like @sfc-gh-japatel was suggested, thanks!

**Comment by sfc-gh-xhuang** [comment]
@sfc-gh-hqin @sfc-gh-japatel @sfc-gh-tzhang Should this be merged such that unshaded versions are released automatically as we go into the GA testing period?

**Review by sfc-gh-tjones — APPROVED** [review]

**Review by sfc-gh-rcheng — APPROVED** [review]

**Comment by sfc-gh-xhuang** [comment]
@sfc-gh-hqin can you resolve conflicts and marge? Thanks!

**Comment by sfc-gh-xhuang** [comment]
@sfc-gh-hqin ping can you help merge ? :)

**Comment by sfc-gh-tzhang** [comment]
This looks to be a duplicate of https://github.com/snowflakedb/snowflake-ingest-java/pull/346, closing

**Pull request closed without merging**
